File: bench/reverse_string.lua
-- BOT-TRANSLATED

local bit = require("bit")

local function string_chars(str)
  local out = {}
  for ind = 1, #str do
    out[ind] = str:sub(ind, ind)
  end
  return out
end

local function reverse(str)
  local low = 1
  local high = #str
  while low < high do
    str[low], str[high] = str[high], str[low]
    low = low + 1
    high = high - 1
  end
end

local function run(str, runs)
  for _ = 1, runs do
    reverse(str)
  end
  reverse(str)
  if table.concat(str) ~= "fedcba9876543210" then
    error("reverse-string output mismatch")
  end
end

run(string_chars("0123456789abcdef"), bit.lshift(1, 25))

File: bench/reverse_string.java
// BOT-TRANSLATED

final class reverse_string {
  static volatile int runs = 1 << 25;

  static void reverse(char[] str) {
    for (int low = 0, high = str.length - 1; low < high; low++, high--) {
      final char tmp = str[low];
      str[low] = str[high];
      str[high] = tmp;
    }
  }

  static void run(char[] str, int runs) {
    while (runs-- > 0) reverse(str);

    reverse(str);

    if (!new String(str).equals("fedcba9876543210")) {
      throw new AssertionError("reverse-string output mismatch");
    }
  }

  public static void main(String[] args) {
    run("0123456789abcdef".toCharArray(), runs);
  }
}
